habitat_baselines/rl/ppo/ppo_trainer.py

```diff
@@ -1035,11 +1036,14 @@ def _eval_checkpoint(
             # NB: Move actions to CPU.  If CUDA tensors are
             # sent in to env.step(), that will create CUDA contexts
             # in the subprocesses.
-            # For backwards compatibility, we also call .item() to convert to
-            # an int
-            if actions[0].shape[0] > 1:
+            if is_continuous_action_space(self.policy_action_space):
+                # Clipping actions to the specified limits
                 step_data = [
-                    action_array_to_dict(self.policy_action_space, a)
+                    np.clip(
```

Comment:
I would like some guidance on this. In order to clam/clip with numpy, I need to have the tensor be a cpu np array. I would prefer to use numpy's clip over torch's clamp because torch only allows for numbers as lower/higher bounds while I would like to use the low and high of the action space (vectors). We can hack around our own version to torch clipping, but I think it would be slightly slower.
I remove some extra `.cpu()` calls, and I am open to ideas on how to make this conversion of torch to numpy as fast as possible.

Reply:
@vincentpierre Seems recent versions of `torch.clamp` support taking in torch.Tensor as the min / max values for the clamp function? https://pytorch.org/docs/1.9.0/generated/torch.clamp.html?highlight=clamp or am I misunderstanding something here? Do we need to bump our min version of PyTorch to solve this issue?

Reply:
Yes, 1.8 does not allow it.